File: 347-top-k-frequent-elements/347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    // count frequency of each number
    const counter = new Map();
    const bucket = [];
    const res = [];
    nums.forEach(num => counter.set(num, (counter.get(num) || 0) + 1));
    for (let [num, frequency] of counter) {
        bucket[frequency] = (bucket[frequency] || new Set()).add(num);
    }
    for (let i = bucket.length - 1; i > 0; i--) {
        if (bucket[i]) res.push(...bucket[i]);
        if (res.length === k) break;
    }
    return res;
};

File: 347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = dict()
        for num in nums:
            if num in counter: counter[num] += 1
            else: counter[num] = 1
        sortedDict = dict(sorted(counter.items(), key=lambda x: x[1]))
        return list(sortedDict)[-k:]
        
        